### PR TITLE
rename 'form' event to 'active_loop'

### LINKED ISSUE #6279: Rename the `form:` event in stories for 2.0
See discussion on Notion [here](https://www.notion.so/rasa/Feedback-on-2-0-4574eaacc0764967ac0b8b82fbc72904#0b5fd7f8b76e4186a1a21c9d821a4958).

A Form is now just a special version of the more general LoopAction, this event in the stories should be renamed to make it clearer what's happening. The current best suggestion is: `active_loop` (open to other suggestions if there are any)

Old version:
image

New version:
image

### PR DESCRIPTION
**Pull request by chkoss**
**Proposed changes**:
Rename `form:` to `active_loop:` in the stories
Fixes #6279

Do **not** rename
- the Form event itself https://github.com/RasaHQ/rasa/blob/cd55d2b08353e0a35cff534113765dafb78d2f83/rasa/core/events/__init__.py#L1117
- the key that is used in markdown stories https://github.com/RasaHQ/rasa/blob/cd55d2b08353e0a35cff534113765dafb78d2f83/rasa/core/training/structures.py#L33

**Note** that we also have to adapt these open PRs:
- [ ] rule policy documentation (#6195)
- [x] example form bot (#6235)

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/core/training/story_reader/yaml_story_reader.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 KEY_USER_INTENT = "intent"
 KEY_SLOT_NAME = "slot"
 KEY_SLOT_VALUE = "value"
-KEY_FORM = "form"
+KEY_FORM = "active_loop"
```
[the remark this answers is not in the document]
Comment (chkoss) [reply rasa/core/training/story_reader/yaml_story_reader.py:28]: I think both is fine. It adds a Form event etc., so can be called `_parse_form`. Do you prefer one of the two terms?

**Comment by chkoss** [comment]
@wochinge Huh, don't know how I missed those occurrences. I did a global search for `form:` on all files and for some reason these two `.yml` files didn't show up. But yeah, searching only `.yml` files makes a lot more sense 😄

File: rasa/core/training/story_reader/yaml_story_reader.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (wochinge) [reply rasa/core/training/story_reader/yaml_story_reader.py:28]: slightly leaning towards `_parse_active_loop` but you're right: as long as the event and so on isn't renamed it doesn't make sense.

[the remark this answers is not in the document]
Comment (chkoss) [reply rasa/core/training/story_reader/yaml_story_reader.py:28]: Okay, then let's rename it if/when we decide to also rename the event etc.

**Comment by chkoss** [comment]
> 👍 We somehow have to make sure that we don't forget the rename in the documentation once we convert that to yaml 🤔 Let's create an issue?

@wochinge Just found this on my to-do list. I think the docs on `master` are already adjusted accordingly, but can you double-check to make sure, please?

**Pull request merged**